py/picca/utils.py

```diff
@@ -76,34 +77,53 @@ def desi_from_truth_to_drq(truth,targets,drq,spectype="QSO"):
 
     '''
 
+    ## Truth table
     vac = fitsio.FITS(truth)
-    vacTargets = fitsio.FITS(targets)
 
-    ## Info of the primary observation
-    thid  = vac[1]["TARGETID"][:]
-    ra    = vacTargets[1]["RA"][:]
-    dec   = vacTargets[1]["DEC"][:]
-    zqso  = vac[1]["TRUEZ"][:]
+    w  = sp.ones(vac[1]["TARGETID"][:].size).astype(bool)
+    print(" start                 : nb object in cat = {}".format(w.sum()) )
+    w &= numpy.core.defchararray.replace(vac[1]["TRUESPECTYPE"][:].astype(str),' ','')==spectype
```

Comment:
@londumas could you simplify this line by doing:

    vac[1]['TRUESPECTYPE'][:].strip()==spectype

Reply:
you won't need to import numpy